lib/services/interact/autoDelegate.ts

```diff
@@ -0,0 +1,46 @@
+import { Trace } from '@voiceflow/base-types';
+
+import { PartialContext, TurnBuilder } from '@/runtime';
+import { Context } from '@/types';
+
+export const MAX_DELEGATION_TURNS = 3;
+
+// mostly just saves us needing to traverse an array twice
+const filterAndGetLastRemovedValue = <A>(list: A[], filterFunc: (a: A) => boolean): [A[], A | null] => {
```

Comment:
not important but typescript convention is to use `T` as the name of generic types

also,
```suggestion
// mostly just saves us needing to traverse an array twice
// TODO: move to @voiceflow/common
const filterAndGetLastRemovedValue = <A>(list: A[], filterFunc: (a: A) => boolean): [A[], A | null] => {
```

or add a jira ticket

Reply:
https://github.com/voiceflow/libs/commit/ee55d9963b2f491696ab8643e78f166b9c6134e3